Revert "Fixup caching" (#16288)
mdn/content: files/en-us/web/http/caching/index.md

...
```

> The cache which stored that response counts the time elapsed since the response was generated as an age. The meaning of `max-age` is that if the age is less than one week, then the response is fresh — and if the age is greater than one week, the the cached response is stale.

Reviewer: ```suggestion
The cache which stored that response counts the time elapsed since the response was generated as an age. The meaning of `max-age` is that if the age is less than one week, then the response is fresh — and if the age is greater than one week, then the cached response is stale.
```

> The cache which stored that response counts the time elapsed since the response was generated as an age. The meaning of `max-age` is that if the age is less than one week, then the response is fresh — and if the age is greater than one week, the the cached response is stale.

Author: ```suggestion
The cache which stored that response counts the time elapsed since the response was generated as an age. The meaning of `max-age` is that if the age is less than one week, then the response is fresh — and if the age is greater than one week, the cached response is stale.
```

Upon receiving that response, the client reverts the stored stale response back to being fresh and can reuse it during the remaining 1 hour.

> The server can can obtain the modification time from the operating-system file system, which is relatively easy to do for the case of serving static files. However, there are some problems; for example, the time format is complex and difficult to parse, and distributed servers have difficulty synchronizing file-update times.

Author: ```suggestion
The server can obtain the modification time from the operating-system file system, which is relatively easy to do for the case of serving static files. However, there are some problems; for example, the time format is complex and difficult to parse, and distributed servers have difficulty synchronizing file-update times.
```

`max-age=0` means that the response is immediately stale, and `must-revalidate` means that it must not be reused without revalidation once it is stale — so in combination, the semantics seem to be the same as `no-cache`.

> However, that usage of `max-age=0` is a remnant of the fact that many implementations prior to HTTP/1.1 were unable to handle the `no-cache` directive — and so to deal with that limitation, `max-age=0` was used as an workaround.

Author: ```suggestion
However, that usage of `max-age=0` is a remnant of the fact that many implementations prior to HTTP/1.1 were unable to handle the `no-cache` directive — and so to deal with that limitation, `max-age=0` was used as a workaround.
```

As a result, a request is validated by `If-None-Match` and `If-Modified-Since`.

> That behavior is also defined in the [Fetch](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch) standard and can be reproduced in JavaScript by calling`fetch()` with the cache mode set to `no-cache` (note that `reload` is not the right mode for this case):

Author: ```suggestion
That behavior is also defined in the [Fetch](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch) standard and can be reproduced in JavaScript by calling `fetch()` with the cache mode set to `no-cache` (note that `reload` is not the right mode for this case):
```

Cache-Control: immutable
```

> ### Primary resources

Author: ```suggestion
### Main resources
```

> Unlike subresources, primary resources cannot be cache busted because their URLs can't be decorated in the same way that subresource URLs can be.

Author: ```suggestion
Unlike subresources, main resources cannot be cache busted because their URLs can't be decorated in the same way that subresource URLs can be.
```

### More about managed caches

> With the method described in previous sections, subresources can be cached for a long time by using cache busting, but primary resources (which are usually HTML documents) can’t be.

Author: ```suggestion
With the method described in previous sections, subresources can be cached for a long time by using cache busting, but main resources (which are usually HTML documents) can’t be.
```

> Caching primary resources is difficult because, using just standard directives from the HTTP Caching specification, there’s no way to actively delete cache contents when content is updated on the server.

Author: ```suggestion
Caching main resources is difficult because, using just standard directives from the HTTP Caching specification, there’s no way to actively delete cache contents when content is updated on the server.
```

However, it is possible by deploying a managed cache such as a CDN or service worker.

> For example, a CDN that allows cache purging via an API or dashboard operation would allow for a more aggressive caching strategy by storing the primary resource and explicitly purging the relevant cache only when an update occurs on the server.

Author: ```suggestion
For example, a CDN that allows cache purging via an API or dashboard operation would allow for a more aggressive caching strategy by storing the main resource and explicitly purging the relevant cache only when an update occurs on the server.
```

A service worker could do the same if it could delete the contents in the Cache API when an update occurs on the server.

> For more information, see the documentation for your CDN, and consult the service worker documentation here.

Author: ```suggestion
For more information, see the documentation for your CDN, and consult [service worker documentation](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API).
```

A service worker could do the same if it could delete the contents in the Cache API when an update occurs on the server.

> For more information, see the documentation for your CDN, and consult [service worker documentation](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API).

Author: ```suggestion
For more information, see the documentation for your CDN, and consult the [service worker documentation](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API).
```
